tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -41,6 +41,8 @@ int main(int argc, char **argv)
 {
     BEGIN_TEST();
     EXPECT_SUCCESS(s2n_disable_tls13());
+
+    /* Test client cipher selection */
```

Comment:
The tests were erroneous previously and had to be moved from its previous location, as `s2n_set_cipher_as_client` was being called on S2N_SERVER conn instead of S2N_CLIENT. I fixed this error and moved all the `s2n_set_cipher_as_client` tests together under `Test client cipher selection `.

Reply:
I'm not worried about the tests you moved, I'm worried about where you moved them to. You combined them with existing tests (line 47-79) intended to check the iana order and number of valid cipher suites, which really doesn't seem related to `s2n_set_cipher_as_client`.

Reply:
I moved them to keep the client side cipher suite tests together and reuse the client connection object and setup. I can add a comment and braces {} around iana order + count test for better clarity.